pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -856,7 +856,17 @@ func TestGarbageCollectorSync(t *testing.T) {
 	stopCh := make(chan struct{})
 	defer close(stopCh)
 	go gc.Run(1, stopCh)
-	go gc.Sync(fakeDiscoveryClient, 10*time.Millisecond, stopCh)
+	// The pseudo-code of GarbageCollector.Sync():
+	// GarbageCollector.Sync():
+	//    wait.Until() loops with `period` until the `stopCh` is closed :
+	//        wait.PollImmediateUntil() loops with 100ms (hardcode) util the `stopCh` is closed:
+	//            controller.WaitForCacheSync() loops with `syncedPollPeriod` (hardcoded to 100ms),
+	//            until either its stop channel is closed after `period`, or all caches synced.
+	//
+	// 200ms of period makes ~2 executions of controller.WaitForCacheSync() in every wait.PollImmediateUntil() loop.
+	// Finally controller.WaitForCacheSync() executes about 10 times (Due to the behavior of time.Ticker,
```

Comment:
Is it really important to run WaitForCacheSync for 10 times? I think the important thing is letting gc.Sync() loop at least twice to ensure the changes made to the fakeDiscoveryClient are picked up.

Reply:
We can't escape from wait.PollImmediateUntil() unless controller.WaitForCacheSync() returns true.
So there are only two paths:
1. gc.Sync() -> wait.PollImmediateUntil() -> controller.WaitForCacheSync() returns true -> wait.PollImmediateUntil() loops.
2. gc.Sync() -> wait.PollImmediateUntil() -> controller.WaitForCacheSync() returns true -> gc.Sync() loops.

Reply:
Done